inputgraph/loader.go

```diff
@@ -498,59 +516,52 @@ func (l *loader) findProject(ctx context.Context) (org, project string, err erro
 			return parts[0], parts[1], nil
 		}
 	}
+
 	return "", "", errors.Wrapf(ErrUnableToDetermineHash, "PROJECT command missing")
 }
 
 func (l *loader) load(ctx context.Context) error {
 	if l.target.IsRemote() {
 		return ErrRemoteNotSupported
 	}
+
 	resolver := buildcontext.NewResolver(nil, nil, l.conslog, "", "", "", 0, "")
-	bc, err := resolver.Resolve(ctx, nil, nil, l.target)
+
+	buildCtx, err := resolver.Resolve(ctx, nil, l.platformResolver, l.target)
 	if err != nil {
 		return err
 	}
-	ef := bc.Earthfile
 
+	l.features = buildCtx.Features
+
+	collOpt := variables.NewCollectionOpt{
+		Console:          l.conslog,
+		Target:           l.target,
+		CI:               l.ci,
+		Push:             l.push,
+		BuiltinArgs:      l.builtinArgs,
+		OverridingVars:   l.overridingVars,
+		EarthlyCIRunner:  l.earthlyCIRunner,
+		PlatformResolver: l.platformResolver,
+		GitMeta:          buildCtx.GitMetadata,
+		Features:         l.features,
+	}
+	l.varCollection = variables.NewCollection(collOpt)
```

Comment:
The `variables.Collection` is being assembled here and reassembled for each child target. Does this look ok?

Reply:
at a quick glance yes. it's unfortunate we need to duplicate this logic from the converter, but it's a necessary evil -- we're going to rely on test-coverage to prevent drift.

Reply:
Great, thanks for confirming.